src/pages/AccountSettings/tabs/Access/SessionsTable.tsx

```diff
@@ -0,0 +1,139 @@
+import {
+  createColumnHelper,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from '@tanstack/react-table'
+import { useCallback, useMemo } from 'react'
+import { useParams } from 'react-router-dom'
+
+import { Session, useDeleteSession } from 'services/access'
+import { cn } from 'shared/utils/cn'
+import { formatTimeToNow } from 'shared/utils/dates'
+import Button from 'ui/Button'
+
+import 'ui/Table/Table.css'
+
+interface SessionColumn {
+  ip: React.ReactNode
+  lastSeen: string
+  userAgent: string
+  revokeBtn: React.ReactNode
+}
+
+const columnHelper = createColumnHelper<SessionColumn>()
+const columns = [
+  columnHelper.accessor('ip', {
+    header: 'IP',
+    cell: (data) => data.renderValue(),
+  }),
+  columnHelper.accessor('lastSeen', { header: 'Last seen' }),
+  columnHelper.accessor('userAgent', { header: 'User agent' }),
+  columnHelper.accessor('revokeBtn', {
+    header: '',
+    cell: (data) => data.renderValue(),
+  }),
+]
+
+interface URLParams {
+  provider: string
+}
+
+interface SessionsTableProps {
+  sessions?: (Session | null)[]
+}
+
+function SessionsTable({ sessions }: SessionsTableProps) {
+  const { provider } = useParams<URLParams>()
+  const { mutate } = useDeleteSession({ provider })
+  const handleRevoke = useCallback(
+    (sessionid: number) => {
+      if (window.confirm('Are you sure you want to revoke this session?')) {
+        mutate({ sessionid })
+      }
+    },
+    [mutate]
+  )
+
+  const data = useMemo(() => {
+    if (!sessions) {
+      return []
+    }
+
+    return sessions.flatMap((s) =>
+      s === null
+        ? []
+        : ({
+            ip: (
+              <p className="w-fit bg-ds-gray-secondary font-mono font-bold text-ds-gray-octonary">
+                {s.ip}
+              </p>
+            ),
+            lastSeen: s.lastseen ? formatTimeToNow(s.lastseen) : '-',
+            userAgent: s.useragent,
+            revokeBtn: (
+              <Button
+                disabled={false}
+                to={undefined}
+                hook="revoke-session"
+                onClick={() => handleRevoke(s.sessionid)}
+                variant="danger"
+              >
+                Revoke
+              </Button>
+            ),
+          } as SessionColumn)
```

Comment:
What is the error here if you remove this cast?

Reply:
Had it there as a type hint to make autocomplete work while filling in the fields. Was pretty sure leaving wouldn't change function nor type-safety, but is there a case I'm missing?

Reply:
One thing you could try instead of casting is to add this as a the return type of the function ... that "should" work and than you're not casting anything here (which is just a way to lie to TS)

Reply:
Yep that worked! Thanks

Reply:
Sick